hm.tex

```diff
@@ -428,7 +430,7 @@ \section{Hamilton's variational principle}\label{sec:varpri}
     \begin{equation}
         \begin{split}
             S(\gamma + h) &= \int_{t_1}^{t_2} L(\gamma + h, \dot \gamma + \dot h, s) \d s \\
-            &= \int_{t_1}^{t_2} 
+            &= \int_{t_1}^{t_2} % not clear how this step happens, specifically where the inner product comes from, maybe a comment on the procedure beforehand
```

Comment:
That is practically mutlivariate Taylor theorem: in R^n these are the Jacobian with respect to the q and qdot variables (so simple row vectors in this case). I was using the scalar product, maybe abusing too much the notation, to emphasize this. Do you think modifying the text above to say `First of all observe that for small $h$, a Taylor expansion gives` (eventually dropping the scalar products) would clarify it enough ?

Reply:
I had guessed that some Taylor business was going on, though I was confused  with the notation indeed. The inner products in the extra terms are new to me, I haven't encountered this approach from my previous courses. I think your modification works good it explicitly mentions Taylor, perhaps also put multivariate to hammer in the idea, that way one has an exact term to google for, in case just "taylor" gives them calculus 1 material.

Reply:
I add the comment on the upcoming version then, thanks!